NavigationReactNative/src/ios/NVNavigationBarView.m

```diff
@@ -19,6 +19,21 @@ - (void)didSetProps:(NSArray<NSString *> *)changedProps
     if ([changedProps containsObject:@"title"]) {
         [self.reactViewController.navigationItem setTitle:self.title];
     }
+    if ([changedProps containsObject:@"barTintColor"]) {
+        [self.reactViewController.navigationController.navigationBar setBarTintColor:self.barTintColor];
+    }
+    if ([changedProps containsObject:@"tintColor"]) {
+        [self.reactViewController.navigationController.navigationBar setTintColor: self.tintColor];
+    }
+    if ([changedProps containsObject:@"titleColor"]) {
+        [self.reactViewController.navigationController.navigationBar setTitleTextAttributes:
+            @{NSForegroundColorAttributeName:self.titleColor}];
```

Comment:
This throws an error if `titleColor` is nil. You can try this out by setting the prop as below and then selecting red followed by orange when on the details screen.
```jsx
titleColor={color === "red" ? color : undefined}
```
You took a different approach on the `NVSceneController`  by setting the `titleTextAttributes` to nil. Although this works it's a bit of a sledgehammer because it will clear out all attributes that the user has set. For example, they might have set the title font using `UIAppearance` (something we want to support because it's hard to do with props alone). 

We can get a mutable copy of the `titleTextAttributes`, just remove our one attribute and then set it back. Whatever approach you take check that it works with `UIAppearance`.
```objective-c
NSMutableDictionary *titleAttributes = [self.reactViewController.navigationController.navigationBar.titleTextAttributes mutableCopy];
[titleAttributes removeObjectForKey:NSForegroundColorAttributeName];
if (self.titleColor != nil) {
    titleAttributes[NSForegroundColorAttributeName] = self.titleColor;
}
[self.reactViewController.navigationController.navigationBar setTitleTextAttributes:titleAttributes];
```

Reply:
yeah thats a better way to do it for sure. I will update and test that out thanks